enterprise/frontend/src/embedding-sdk/components/private/SdkContext.tsx

```diff
@@ -0,0 +1,20 @@
+import type React from "react";
+import type { ReactNode } from "react";
+import { createContext, useContext } from "react";
+
+const SdkContext = createContext<boolean>(false);
+
+export const SdkContextProvider = ({ children }: React.PropsWithChildren) => {
+  return <SdkContext.Provider value={true}>{children}</SdkContext.Provider>;
+};
+
+export const useIsInSdkProvider = () => {
+  return useContext(SdkContext);
+};
```

Comment:
Not sure I'm following, `useIsInSdkProvider` only returns if the component is inside of the Metabaseprovider, I guess we could wrap useSelector with a try catch but it'd be a bit weird.
I'm using this hook only to stop rendering the three below in case we're outside of the provider as that will make the host app in error. Unfortunately next (but possibly other framework with lazy loading) will render some components without the provider for a brief moment when hot reloading and with this approach it'll still work.

I do agree that we should take components out of the redux state

Reply:
@npretto Sorry, I thought that there was a place where we used that hook for detecting whether we're in the SDK's context or not, but I couldn't find it anymore. I must be daydreaming...

Reply:
No worries, without checking what the value is, it's easy to think it's something more complex than a boolean